src/mym.cpp

```diff
@@ -1800,3 +1808,83 @@ mxArray* deserialize(const char* rpSerial, const size_t rlength) {
         mxFree(p_cmp);
     return p_res;
 }
+char *hex2char(char *original_val, const size_t char_length) {
+    const uint8_t *pnt = (uint8_t *)original_val;
+    uint8_t *result_pnt = new uint8_t[char_length*4];
+    unsigned int offset = 0;
+    for( unsigned int a = 0; a < char_length; ++a )
+    {
+        if     (pnt[a]<=0x7F) {
+            result_pnt[a+offset] = pnt[a];
+        }
+        else if(pnt[a]<=0x7FF) {
+            result_pnt[a+offset] = ((pnt[a]>>6) + 0xC0);
+            result_pnt[a+offset+1] = ((pnt[a] & 0x3F) + 0x80);
+            offset += 1;
+        }
+        else if(0xD800<=pnt[a] && pnt[a]<=0xDFFF) {
+            mexErrMsgIdAndTxt("DataJoint:Deserialization:UTF8",
+                "Invalid block of UTF8 detected.");
+        } //invalid block of utf8
+        else if(pnt[a]<=0xFFFF) {
+            result_pnt[a+offset] = ((pnt[a]>>12) + 0xE0);
+            result_pnt[a+offset+1] = (((pnt[a]>>6) & 0x3F) + 0x80);

```

Comment:
a+offset+1 > char_length

Reply:
Yes, this is correct but I am populating and returning `result_pnt`. It is by definition larger in length from argument `original_val` since it represents the HEX bytes. See:
``` cpp
uint8_t *result_pnt = new uint8_t[char_length*4];
```
It has been pre-allocated with the maximum number of elements for a character length (per UTF-8).

Reply:
Ahh missed this. think we're good.